tests/modal.ts

```diff
@@ -0,0 +1,1097 @@
+        out_mod.show();
+        expect(out_mod.parentElement).to.equal(document.body);
+        const in_mod = out_mod.querySelector("#inner_modal");
+        const in_par = in_mod.parentElement;
+        out_mod.animationDur = 0;
+        in_mod.show();
+        expect(out_mod.parentElement).to.equal(out_par);
+        expect(in_mod.parentElement).to.equal(document.body);
+
+        const stack_modal = el1.querySelector("#stack_modal");
+        out_mod.show();
+        stack_modal.show();
+        stack_modal.animationDur =0;
+        expect(stack_modal.parentElement).to.equal(out_mod.parentElement);
+        expect(stack_modal.parentElement).to.equal(document.body);
+        expect(out_mod.parentElement).to.equal(document.body);
+    });
+
+    it('closesel', async () => {
+        const el1 = await fixture (html`
+            <div id = "parent">
+                <button id="open_btn"></button>
+                <aalam-modal id="modal" closesel=".closing">
+                       <div id="slot" slot="modal-body">
+                           <button id = "close_btn"></button>
+                           <button id="new_button"></button>
+                       </div>
+                </aalam-modal>
+            </div>
+        `);
+        const modal = el1.querySelector("#modal");
+        modal.animationDur = 0;
+        const slot_id = el1.querySelector("#slot");
+        const modal_par = modal.parentElement;
+        modal.show();
+        const new_btn = modal.querySelector("#new_button");
+        const new_el=document.createElement("button");
+        new_el.classList.add("closing");
+        slot_id.appendChild(new_el);
+        const new_el_wrap = new_el.getBoundingClientRect();
+        expect(modal.parentElement).to.equal(document.body);
+        new_el.click();
+        expect(modal.parentElement).to.equal(modal_par);
+    });
+
+    it('noesc', async () => {
+        const el1 = await fixture(html`
+            <div id="parent">
+                <aalam-modal id="modal">
+                    <div slot="modal-body"></div>
+                </aalam-modal>
+            </div>
+        `);
+        const modal = el1.querySelector("#modal");
+        const modal_parent = modal.parentElement;
+        modal.show();
+        modal.animationDur = 0;
+        expect(modal.parentElement).to.equal(document.body);
+        await sendKeys({down:"Escape"});
+        expect(modal.parentElement).to.equal(modal_parent);
+
+        const el2 = await fixture(html `
+            <div id="parent">
+                <aalam-modal id="modal" noesc>
+                    <div slot="modal-body"></div>
+                </aalam-modal>
+            </div> `);
+        const el_modal = el2.querySelector("#modal");
+        el_modal.show();
+        el_modal.animationDur = 0;
+        expect(el_modal.parentElement).to.equal(document.body);
+        await sendKeys({down:"Escape"});
+        expect(el_modal.parentElement).to.equal(document.body);
```

Comment:
Try to setAttribute to el1 with noesc attribute and then send the esc keys, it should not close. Also remoteAttribute for el2 and check esc is working.